RadixWallet/Features/TransactionReviewFeature/TransactionReview+Sections.swift

```diff
@@ -934,3 +934,93 @@ extension TransactionReview.ResourceInfo {
 		}
 	}
 }
+
+extension [String: [ResourceIndicator]] {
+	/// Aggregate the transfer amounts for the same resource for the same account.
+	///
+	/// The RET analysis might return multiple withdrawls/deposits for the same resource,
+	/// instead of showing separate entries for each withdral/deposit, we aggregate
+	/// the fungible amounts or the non fungible ids.
+	///
+	/// Important note: This aggregates only the guranteed amounts,
+	/// predicted amounts cannot be aggregated since each amount will have a specific instruction index attached.
+	///
+	/// This function is only used curently for stake claim transactions, when the Dapp might sent a manifest
+	/// which is interpreted by RET as having multiple distinct withdrawls/deposits for the same resource.
+	///
+	/// This should eventually be moved to RET, so it should return the aggregated amounts
+	var aggregated: Self {
+		var aggregatedResult: Self = [:]
+
+		for (key, value) in self {
+			var result: [ResourceIndicator] = []
+			for indicator in value {
+				if let i = result.firstIndex(where: {
+					$0.resourceAddress == indicator.resourceAddress &&
+						$0.isGuaranteedAmount &&
+						indicator.isGuaranteedAmount
+				}) {
+					result[i].add(indicator)
+				} else {
+					result.append(indicator)
+				}
+			}
+
+			aggregatedResult[key] = result
+		}
+		return aggregatedResult
+	}
+}
+
+extension ResourceIndicator {
+	var isGuaranteedAmount: Bool {
+		switch self {
+		case .fungible(_, .guaranteed):
+			true
+		case .nonFungible(_, .byIds):
+			true
+		default:
```

Comment:
is it not feasible to enumerate all cases? I tend to dislike `default` usage (which has bitten us before).

Reply:
I would agree. The trues and the falses can be aggregated so it still only uses two rows